pkg/ingester/stream.go

```diff
@@ -346,7 +348,7 @@ func (s *stream) validateEntries(entries []logproto.Entry, isReplay bool) ([]log
 		}
 
 		now := time.Now()
-		if !s.cfg.RateLimitWholeStream && !s.limiter.AllowN(now, len(entries[i].Line)) {
+		if rateLimitWholeStream && !s.limiter.AllowN(now, len(entries[i].Line)) {
```

Comment:
This ought to be `!rateLimitWholeStream`. 

I see why our test missed this: because whole-stream rate limiting happens at the end, the result looks the same. It's important that we not call `AllowN` when `rateLimitWholeStream==true` because a successful call changes the limiter's state and affects the rate limit. It there a way we can tell how the limiter was called in the test to help us in the future?

Reply:
great catch, thank you. I'll investigate how to do that.

Reply:
Pushed a fix and a test covering it, wdyt?